blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -150,20 +156,84 @@ class BlazeServerBuilder[F[_]](
       maxHeadersLen: Int = maxHeadersLen): Self =
     copy(maxRequestLineLen = maxRequestLineLen, maxHeadersLen = maxHeadersLen)
 
+  @deprecated("Use withSSLContext and withTLSParameters instead", "0.21.0-RC3")
+  @silent("deprecated")
   def withSSL(
       keyStore: StoreInfo,
       keyManagerPassword: String,
       protocol: String = "TLS",
       trustStore: Option[StoreInfo] = None,
       clientAuth: SSLClientAuthMode = SSLClientAuthMode.NotRequested): Self = {
-    val bits = KeyStoreBits(keyStore, keyManagerPassword, protocol, trustStore, clientAuth)
-    copy(sslBits = Some(bits))
+
+    val sslContext = F.delay {
+      val ksStream = new FileInputStream(keyStore.path)
+      val ks = KeyStore.getInstance("JKS")
+      ks.load(ksStream, keyStore.password.toCharArray)
+      ksStream.close()
+
+      val tmf = trustStore.map { auth =>
+        val ksStream = new FileInputStream(auth.path)
+
+        val ks = KeyStore.getInstance("JKS")
+        ks.load(ksStream, auth.password.toCharArray)
+        ksStream.close()
+
+        val tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
+
+        tmf.init(ks)
+        tmf.getTrustManagers
+      }
+
+      val kmf = KeyManagerFactory.getInstance(
+        Option(Security.getProperty("ssl.KeyManagerFactory.algorithm"))
+          .getOrElse(KeyManagerFactory.getDefaultAlgorithm))
+
+      kmf.init(ks, keyManagerPassword.toCharArray)
+
+      val context = SSLContext.getInstance(protocol)
+      context.init(kmf.getKeyManagers, tmf.orNull, null)
+      context.some
+    }
+
+    copy(sslContext = sslContext, sslEngineConfig = clientAuthModeToTlsParameters(clientAuth))
   }
 
+  def withSSLContext(sslContext: SSLContext): Self =
+    copy(sslContext = F.pure(sslContext.some))
+
+  @deprecated(
+    "Overload with clientAuth param is deprecated.  Use withSSLContext(SSLContext).withTLSParameters(TLSParameters(...)) instead.",
+    "0.21.0-RC3")
+  @silent("deprecated")
   def withSSLContext(
       sslContext: SSLContext,
       clientAuth: SSLClientAuthMode = SSLClientAuthMode.NotRequested): Self =
-    copy(sslBits = Some(SSLContextBits(sslContext, clientAuth)))
+    copy(
+      sslContext = F.pure(sslContext.some),
+      sslEngineConfig = clientAuthModeToTlsParameters(clientAuth))
+
+  def withoutSSLContext: Self =
+    copy(sslContext = F.pure(None))
+
+  def withSSLParameters(sslParameters: SSLParameters): Self =
```

Comment:
This is useful over `TLSParameters` because it can be created from a default `SSLParameters` for whatever `SSLContext` is on hand.  `TLSParameters` needs to be configured from _its_ defaults, or we'd need to copy an `SSLParameters` to `TLSParameters` only to copy it back.

This effectively makes `TLSParameters` the convenience method.